### ISSUE TITLE
Generate nothing from VLLM output

### ISSUE THREAD

**Issue by FocusLiwen** [vllm-project/vllm#1185]
When I run batch inferences, sometimes, the output from vLLM is empty, meaning prediction is empty. Could we make it at least it generate one token? The output is empty is also strange.

**Comment by AnupKumarJha** [comment]
@FocusLiwen can you add some more detail like how you are using inferencing , sampling params & what is your request

**Comment by FocusLiwen** [comment]
Hi, I used tensor_parallel_size as 2 with the following parameters:
"max_tokens": 128,
"temperature": 0,
"top_p": 1.0,
"top_k": -1
I also extract the output from the generation call:

"gold": {"text": "Sports", "supplements": {}}, 
"predictions": [{"text": "", "raw_text": "", "logprob": 0, "tokens": []}]}

**Comment by FocusLiwen** [comment]
In Huggingface generation api, there is a parameter called: min_gen_len, which could be set as 1 to avoid zero output. But with vLLM, there is no such parameters.

**Comment by viktor-ferenczi** [comment]
Does this happen if you increase the temperature to 1e-3 or 1e-2?

**Comment by summer66** [comment]
> In Huggingface generation api, there is a parameter called: min_gen_len, which could be set as 1 to avoid zero output. But with vLLM, there is no such parameters.

We ran into the same problem (no output). Being able to set a minimal length for the generated text will be very helpful.

**Comment by ms-maoquan** [comment]
meet the same issue. when run batch inferences, sometimes there are empty output. But change it to run by single-case, the reuslt  is not empty.

**Comment by raihan0824** [comment]
> The [Mixtral AWQ vllm](https://huggingface.co/TheBloke/Mixtral-8x7B-Instruct-v0.1-AWQ) example gives empty output (with temperature 0,0.5,1.0 or default sampling parameters).

same, have you solve this?

**Comment by Andrew-MAQ** [comment]
> The [Mixtral AWQ vllm](https://huggingface.co/TheBloke/Mixtral-8x7B-Instruct-v0.1-AWQ) example gives empty output (with temperature 0,0.5,1.0 or default sampling parameters).



> > The [Mixtral AWQ vllm](https://huggingface.co/TheBloke/Mixtral-8x7B-Instruct-v0.1-AWQ) example gives empty output (with temperature 0,0.5,1.0 or default sampling parameters).
> 
> same, have you solve this?

This is the same for me as well. It says 1024 completion_tokens, but the content is blank.

**Comment by sfc-gh-hazhang** [comment]
It is likely your weights are corrupted

**Comment by HIT-Owen** [comment]
> The [Mixtral AWQ vllm](https://huggingface.co/TheBloke/Mixtral-8x7B-Instruct-v0.1-AWQ) example gives empty output (with temperature 0,0.5,1.0 or default sampling parameters).

same problem, has it been resolved?

**Comment by imadcap** [comment]
Same issue there!
using vllm==0.3.0+cu118 with TheBloke/Mixtral-8x7B-Instruct-v0.1-AWQ
something is definitely wrong
if anyone knows why

**Comment by hahmad2008** [comment]
same here with mistral the output is empty

**Comment by Meersalzeis** [comment]
I ecountered the same issue, empty string text output, with the [TheBloke Mixtral AWQ](https://huggingface.co/TheBloke/Mixtral-8x7B-Instruct-v0.1-AWQ) with vllm and 2 loaders from [Oobaboogas Web UI](https://github.com/oobabooga/text-generation-webui) as well. However, ybelkada/Mixtral-8x7B-Instruct-v0.1-AWQ worked on both vLLM and the WebUI for me.

I'm still not 100% sure its a faulty model, so I'd be happy If one of you can confirm (or deny) this with your setup.

**Comment by cieske** [comment]
In my case using Mistral Instruct model, input with proper template and setting max_tokens from SamplingParams helps.

**Comment by sAviOr287** [comment]
Are there any news on this. I also get empty strings when running in batches returning the empty string for some reason. If using batch size of one this never happens. Any update on this would be great.

**Comment by AmoghM** [comment]
Same issue!

**Comment by hugocool** [comment]
im facing the same issue with llama 3 8b on a 48 gb vram gpu while using the outlines library to enforce json responses, the fields are empty, even though there is plenty of ram left and the model is loaded on the gpu completely.

**Comment by joaograndotto** [comment]
same problem, when the prompt is run for the first time it generates normally, if run again the same prompt returns an empty response

**Comment by zichaow** [comment]
encountered the same issue

**Comment by fc2869** [comment]
For some reason, setting `min_tokens=1` did not work for me, but `min_tokens=2` worked.

**Comment by ronchengang** [comment]
same issue here.

**Comment by ShengGuanWSU** [comment]
same issue here, changing min_tokens won't help my case.

**Comment by av** [comment]
Experienced this today, runnin a somewhat exotic GPTQ quant: [ChenMnZ/Mistral-Large-Instruct-2407-EfficientQAT-w2g64-GPTQ](https://huggingface.co/ChenMnZ/Mistral-Large-Instruct-2407-EfficientQAT-w2g64-GPTQ)

<details><summary>Sample request</summary>
<p>

```bash
curl {{host}}/v1/chat/completions -H 'Content-Type: application/json' -H "Authorization: Bearer ---" -d '{
  "model": "ChenMnZ/Mistral-Large-Instruct-2407-EfficientQAT-w2g64-GPTQ",
  "messages": [
    {
      "role": "user",
      "content": "Answer in one word. Where is Paris?"
    }
  ],
  "min_tokens": 2,
  "max_tokens": 8
}'
```

</p>
</details> 


<details><summary>Sample response</summary>
<p>

```json
{
  "id": "chat-9e94785f8fd94d1d93be478cb96ddaea",
  "object": "chat.completion",
  "created": 1722955194,
  "model": "ChenMnZ/Mistral-Large-Instruct-2407-EfficientQAT-w2g64-GPTQ",
  "choices": [
    {
      "index": 0,
      "message": {
        "role": "assistant",
        "content": "",
        "tool_calls": []
      },
      "logprobs": null,
      "finish_reason": "length",
      "stop_reason": null
    }
  ],
  "usage": {
    "prompt_tokens": 12,
    "total_tokens": 20,
    "completion_tokens": 8
  }
}
```

</p>
</details> 

Engine args
```bash
--cpu-offload-gb 26 --max-model-len 1024 --enforce-eager
```

I've tried many other combinations, however they didn't work either

**Comment by drizzle0171** [comment]
same issue here

**Comment by jhealy1** [comment]
Same issue here, I notice when I make rapid invocations the return tends to be blank but that `sleep(a few seconds)` between invocations greatly improves my odds of getting non blank return

**Comment by satyamraj18** [comment]
Yeah, I also faced the same issue. Adding min_tokens was something which helped me.

**Comment by Mace1059** [comment]
I am having the same problem.

**Comment by YangQiangli** [comment]
I am having the same problem.

**Comment by Raibows** [comment]
I had the same issue but I found that it was because the input prompt exceeded the max length of the model. By manually truncating the input prompt, it started to work.

**Comment by HomingHamster** [comment]
notably different from the openai api is that the max_tokens includes the prompt whereas the option on the openai api does not seem to include the prompt

**Comment by Carol-gutianle** [comment]
setting min_tokens is the quickest solution to this problem.
